src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -0,0 +1,140 @@
+package seedu.address.model.session;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+/**
+ * Class that handles RecurringSession that extend Session.
+ */
+public class RecurringSession extends Session {
+    public static final String MESSAGE_CONSTRAINTS = "End date/time does not match starting date and interval.";
+    private Interval interval;
+    private SessionDate endDateTime;
+
+    /**
+     * Constructs a {@code RecurringSession}.
+     *
+     * Requires that all fields entered to be non null.
+     * @param sessionDate
+     * @param duration
+     * @param subject
+     * @param fee
+     * @param interval
+     * @param endDateTime must match sessionDate and interval.
+     */
+    public RecurringSession(SessionDate sessionDate, Duration duration, Subject subject, Fee fee,
```

Comment:
Also I am just wondering if there could be a way (Probably through a command) to add a recurring session into to application, so that the reviewers can checkout and run the PR to perform manual testing as well, to better understand your implementation. Not sure if you want to work on that first before we reveal and approve this to ensure that the implementation passes some light testing. If not it might be hard for me and jonah to test if our feature works if we couldn't add recurring session inside to the addressbook to test.

Reply:
yep will do this